xunit.runner.visualstudio.testadapter/VsTestRunner.cs

```diff
@@ -363,9 +369,7 @@ void RunTests(IRunContext runContext, IFrameworkHandle frameworkHandle, LoggerHe
             var assemblyFileName = runInfo.AssemblyFileName;
             var assemblyDisplayName = Path.GetFileNameWithoutExtension(assemblyFileName);
             var shadowCopy = assembly.Configuration.ShadowCopyOrDefault;
-            var appDomain = assembly.Configuration.AppDomain ?? AppDomainDefaultBehavior;
-            if (DisableAppDomainRequestedInRunContext(runContext.RunSettings.SettingsXml))
-                appDomain = AppDomainSupport.Denied;
+            var appDomain = !RunSettingsHelper.DisableAppDomain ? AppDomainSupport.Denied : (assembly.Configuration.AppDomain ?? AppDomainDefaultBehavior);
```

Comment:
I'm also not a fan of collapsing the two lines here. I prefer it the way it was before, because the readability is too low with there being three possible outcomes.